### PR TITLE
Content/multi model pipelines

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Pull request for deliverable 4 (multi-model pipelines) which includes

- New content on the landing page 

- Example notebook for multi-model pipelines

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by PhilippeMoussalli** [comment]
@yeandy @rezarokni

**Review by damccorm — COMMENTED** [review]
The diff won't render so I can't leave comments inline on the notebook unfortunately. Mostly, this LGTM, my only comments are:

1) In the Model handlers section

> The key is used o keep a reference

Should be `The key is used to keep a reference

2) Could you please add the comment about issue https://github.com/apache/beam/issues/21863 to CLIPWrapper and PytorchNoBatchModelHandlerKeyedTensor

3) Could you please fix the whitespace precommit violations and resolve the merge conflicts?

@yeandy I know you had thoughts about using cascade vs ensemble, does that look right in this PR? Also feel free to drop in other comments of course :)

File: website/www/site/content/en/documentation/ml/multi-model-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+## How to build a Multi-model pipeline with Beam
+
+A typical machine learning workflow involves a series of data transformation steps such as data
+ingestion, data processing tasks, inference, and post-processing. Beam enables you to orchestrate
+all of those steps together by encapsulating them in a single Beam DAG. This allows you to build
+resilient and scalable end-to-end machine learning systems.
+
+To deploy your machine learning model in a Beam pipeline, you can use
+the [`RunInferenceAPI`](https://beam.apache.org/documentation/sdks/python-machine-learning/) which
```
Comment (yeandy) [inline website/www/site/content/en/documentation/ml/multi-model-pipelines.md:37]: Can we also change the documentation on this [page](https://github.com/apache/beam/blob/master/website/www/site/content/en/documentation/sdks/python-machine-learning.md#ensemble-pattern) to match the wording here? I think I found 2 instances of the word "ensemble" in the other page.

**Comment by PhilippeMoussalli** [comment]
> The diff won't render so I can't leave comments inline on the notebook unfortunately. Mostly, this LGTM, my only comments are:
> 
> 1. In the Model handlers section
> 
> > The key is used o keep a reference
> 
> Should be `The key is used to keep a reference
> 
> 2. Could you please add the comment about issue #21863 to CLIPWrapper and PytorchNoBatchModelHandlerKeyedTensor
> 3. Could you please fix the whitespace precommit violations and resolve the merge conflicts?
> 
> @yeandy I know you had thoughts about using cascade vs ensemble, does that look right in this PR? Also feel free to drop in other comments of course :)

thanks for reviewing @damccorm. I resolved the issues mentioned above.

**Review by yeandy — APPROVED** [review]
Let's use "cascade". So we need to also update the wording in the other link (see comment below). Notebook looks good. Thanks @PhilippeMoussalli and @damccorm!

File: website/www/site/content/en/documentation/ml/overview.md [hunk 2]
```diff
@@ -58,4 +58,5 @@ In order to automate and track the AI/ML workflows throughout your project, you
 ## Examples
 
 You can find examples of end-to-end AI/ML pipelines for several use cases:
+* [Multi model pipelines in Beam](/documentation/ml/online-clustering)
```
Comment (Juta) [inline website/www/site/content/en/documentation/ml/overview.md:61]: this one is linking to the wrong resource

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**